app/routes/task_routes.py

```diff
@@ -1 +1,144 @@
-from flask import Blueprint
\ No newline at end of file
+from datetime import datetime
+
+from flask import request, Blueprint, make_response, abort
+from sqlalchemy import asc, desc
+
+from app import db
+from app.models.task import Task
+from app.slack.slack_client import SlackClient
+from app.slack.slackmessage import SlackMessage
+
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+slack_client = SlackClient()
+
+
+@tasks_bp.post("")
+def create_task():
+    request_body = request.get_json()
+    if "title" not in request_body.keys():
+        abort(make_response({"details": "Invalid data"}, 400))
+    title = request_body["title"]
+    if "description" not in request_body.keys():
+        abort(make_response({"details": "Invalid data"}, 400))
+    description = request_body["description"]
+    new_task = Task(title=title, description=description, completed_at=None)
+    db.session.add(new_task)
+    db.session.commit()
+    response = {
+        "task": new_task.to_dict()
+    }
+    return response, 201
+
+
+@tasks_bp.get("")
+def get_all_tasks():
+    description_param = request.args.get("description")
+    title_param = request.args.get("title")
+    sort_param = request.args.get("sort")
+
+    query = db.select(Task)
+    if description_param:
+        query = query.where(Task.description.like(f"%{description_param}%"))
+
+    if title_param:
+        query = query.where(Task.title.like(f"%{title_param}"))
+    if sort_param == "asc":
+        query = query.order_by(asc(Task.title))
+    elif sort_param == "desc":
+        query = query.order_by(desc(Task.title))
+
+    tasks = db.session.scalars(query).all()
+    tasks_response = [task.to_dict() for task in tasks]
+    return tasks_response
+
+
+def validate_id(id):
+    if not id.isnumeric():
+        abort(make_response({"message": f"Tasks id {id} invalid"}, 400))
+
```

Comment:
I'd expect the logic for a method like `validate_id` to check that an ID is valid in the database too, not just check that `id` is an integer. If you have both of these pieces of logic in this method and also put your error handling in this method, you would be able to DRY up many of your routes quite a bit.